docs/dev-notes/DEVELOPMENT.md

```diff
@@ -0,0 +1,135 @@
+# Development
+
+## Getting Started
+
+To get started with the project, you might need several tools to compile on your machine.
+
+- First install IDE or Code Editor you like so you can navigate to code conveniently
+  - [Visual Studio Code](https://code.visualstudio.com/) (Recommended) - Code Editor to navigate between project files
+  - [Android Studio](https://developer.android.com/studio) and [XCode](https://developer.apple.com/xcode/) (Required) - IDEs for setting up CLI tools and getting suggestions on native code
+- Setup react native development environment on your machine by following commands below
+
+> This guide assumes you have [homebrew](https://brew.sh/) installed and combines most of the commands you would need to setup the environment on your machine. If you want to read full guide, visit the [official docs](https://reactnative.dev/docs/environment-setup) -> React Native CLI Quickstart -> Your Development and Target OS
+
+  - Install node using `brew install node` (Node 14 or newer required, prefer LTS versions)
+  - Check if you have ruby installed on your machine by running `ruby --version` (React Native uses [this version](/.ruby-version) of Ruby)
+
+> Currently, macOS 12.5.1 is shipped with Ruby 2.6.8, which is not what is required by React Native. We recommend installing some Ruby Manager to have different versions of Ruby installed on your machine without breaking other projects. [React Native docs have listed a few](https://reactnative.dev/docs/environment-setup#ruby), you can choose any of them and skip instructions for Ruby Version Manager if you want to go with a different option.
+
+ - If you do not have required version of ruby installed
+   - Install [rbenv](https://github.com/rbenv/rbenv) for managing Ruby versions by running `brew install rbenv ruby-build`
+   - Load rbenv in your shell by running `rbenv init` and restarting the shell
+   - Make sure you have the required ruby version set locally. You can do this by running `rbenv local 2.7.6`
```

Comment:
I am a big fan of `nvm` and `rbenv` to install and manage node/ruby versions on my machine. I think it makes life much easier for everyone and avoids compilation issues sometimes. 

I suggest that instead of having docs in here saying, "check what version of ruby you use, macos uses X version", we just give instructions on how to use rbenv. Just assume everyone on the team is using it. 

I think we can do the same with `nvm` to have everyone use a certain version of node on the project. Although, node is quite backwards compatible so using version 14, 15, 16, etc might still compile the project just fine?

Reply:
The instructions next to `If you do not have required version of ruby installed` actually guides how to setup `rbenv`. I just wanted to give option so anyone using any other Ruby Version Manager can still use the same to set the correct ruby version. However, our recommendation is to use `rbenv`. The instructions may be slightly confusing? I can re-write them to make more clear.

For node,  you are correct. It is quite backwards compatible e.g. React Native requires 14+ and I'm using `v16.19.0`, but it works fine. So node won't create much issues. But I'll see if we can add instructions for `nvm`, it would be even better.